src/shared/components/query/StudySearch.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+    addClauses,
+    removePhrase,
+    toQueryString,
+} from 'shared/lib/query/textQueryUtils';
+import * as React from 'react';
+import { FunctionComponent, useState } from 'react';
+import {
+    SearchClause,
+    QueryUpdate,
+} from 'shared/components/query/filteredSearch/SearchClause';
+import { QueryParser } from 'shared/lib/query/QueryParser';
+import _ from 'lodash';
+import { SearchBox } from 'shared/components/query/filteredSearch/SearchBox';
+import { StudySearchControls } from 'shared/components/query/filteredSearch/StudySearchControls';
+
+export type AutosuggestStudySearchProps = {
+    parser: QueryParser;
+    query: SearchClause[];
+    onSearch: (query: SearchClause[]) => void;
+};
+
+export const StudySearch: FunctionComponent<AutosuggestStudySearchProps> = function(
+    props
+) {
+    const [isMenuOpen, setMenuOpen] = useState(false);
```

Comment:
for consistency, i would like to use mobx for handling all state, local or global.  look up useLocalObservable for the analog to useState hook.  it's already used a bit in our codebase.

Reply:
Check, fixed in [d8627b7](https://github.com/cBioPortal/cbioportal-frontend/pull/4305/commits/d8627b7399cf5c4e0a6e32decfb6d7d1e2a15ace)